### PR TITLE
Add linear API in FeasibilityFormNLS

### PR DESCRIPTION
**Pull request by tmigot**
Used in https://github.com/JuliaSmoothOptimizers/NLPModelsKnitro.jl/pull/88

### CODE REVIEW

**Comment by tmigot** [comment]
I realized there were some tests missing for the linear API, and then some issues in the current implementation. Since, it is quite a large PR here is a recap':
- [ ] `test/nls/simple-model.jl` I added a linear constraint to the test problem.
- [ ] This has impact on `test/nls/slack-model.jl`
- [ ] I updated the show test in `test/nls/feasibility-residual.jl` and `test/nls/feasibility-form-nls.jl`
- [ ] I added the linear API for `FeasibilityFormNLS`
- [ ] For `FeasibilityFormNLS`, we don't count the linear constraints from the residual as it is not implemented.
- [ ] I had to fix the implementation of `SlackModel`

**Comment by tmigot** [comment]
The breaking packages are the one, where the linear API is not implemented.

File: src/feasibility-form-nls.jl [hunk 1]
```diff
@@ -117,76 +119,116 @@ function NLPModels.objgrad!(nlp::FeasibilityFormNLS, x::Array{Float64}, g::Array
...
-  @lencheck nlp.meta.nnzj rows cols
-  n, m, ne = nlp.internal.meta.nvar, nlp.internal.meta.ncon, nlp.internal.nls_meta.nequ
+  @lencheck nlp.meta.nln_nnzj rows cols
+  n, m, ne = nlp.internal.meta.nvar, nlp.internal.meta.nnln, nlp.internal.nls_meta.nequ
   nnzjF = nlp.internal.nls_meta.nnzj
   @views jac_structure_residual!(nlp.internal, rows[1:nnzjF], cols[1:nnzjF])
   if m > 0
-    idx = nnzjF .+ (1:(nlp.internal.meta.nnzj))
-    @views jac_structure!(nlp.internal, rows[idx], cols[idx])
+    idx = nnzjF .+ (1:(nlp.internal.meta.nln_nnzj))
```
Comment (abelsiqueira) [inline src/feasibility-form-nls.jl:155]:
```suggestion
    idx = nnzjF .+ 1:nlp.internal.meta.nln_nnzj
```

File: src/slack-model.jl [hunk 2]
```diff
@@ -1,6 +1,29 @@
 export SlackModel, SlackNLSModel
 
-get_slack_ind(jl, ind) = map(b -> findfirst(x -> x == b, ind), intersect(jl, ind))
+function get_slack_ind(jl, ind)
+  if intersect(jl, ind) == []
+    return Int[]
+  end
+  return map(b -> findfirst(x -> x == b, ind), intersect(jl, ind))
+end
```
Comment (abelsiqueira) [inline src/slack-model.jl:8]: See #61

**Review by abelsiqueira — APPROVED** [review]
I made a small suggestion, but that is not blocking. Thanks for the PR.

File: src/slack-model.jl (hunk 2 above)
Comment (tmigot) [reply src/slack-model.jl]: I rebased.

**Pull request merged**